libexec/pyenv-rehash

```diff
@@ -6,32 +6,58 @@ set -e
 
 SHIM_PATH="${PYENV_ROOT}/shims"
 PROTOTYPE_SHIM_PATH="${SHIM_PATH}/.pyenv-shim"
+LOCK_FILE_PATH="${SHIM_PATH}/.pyenv.lock"
+LOCK_FILE_FD=90
```

Comment:
At least I don't think it's good idea to hardcode `fd` of lock file in the script....